Sources/BrowserServicesKit/ContentBlocking/UserScripts/TrackerResolver.swift

```diff
@@ -132,6 +133,21 @@ public class TrackerResolver {
         
         return blockingState
     }
+
+    private func findMatchingRuleWithAction(tracker: KnownTracker,
+                                            trackerUrlString: String,
+                                            resourceType: String,
+                                            pageUrlString: String) -> (KnownTracker.Rule, TrackerResolver.RuleAction)? {
+        guard let host = URL(string: pageUrlString)?.host else { return nil }
+        var ruleIndex = 0
+        while let rule = tracker.ruleMatchingUrl(trackerUrlString, startingFrom: ruleIndex) {
+            if let action = rule.action(type: resourceType, host: host) {
+                return (rule, action)
+            }
+            ruleIndex = tracker.rules!.firstIndex(of: rule)! + 1
+        }
```

Comment:
I wonder if we can make this logic more easy to follow, as it branches in few places there (and in called methods).

My main concerns is how we operate on indexes, to then find index and retry from position - maybe we can just refactor this into `tracker.rules.reversed()` and `popLast()` in a loop (or equivalent with `removeFirst()`, but that requires `isEmpty` check)?